### ISSUE TITLE
Issue with training ShapeNet.py

### ISSUE THREAD

**Issue by edshkim98** [XuyangBai/KPConv.pytorch#5]
Hi, thank you for releasing your code!
I have tried to run your code for shapenet part segmentation, but whenever I run the code, the code seems to get stuck on the
same part every time.
When I run **python training_ShapeNetPart.py**, the output message appears normally as:
**Calib Neighbors 00000065: timings 5.71s
Calib Neighbors 00000067: timings 5.88s
Calib Neighbors 00000069: timings 6.04s
Calib Neighbors 00000070: timings 6.15s
Calib Neighbors 00000072: timings 6.31s
Calib Neighbors 00000073: timings 6.43s
Calib Neighbors 00000075: timings 6.57s
Calib Neighbors 00000077: timings 6.75s
Calib Neighbors 00000078: timings 6.85s
Calib Neighbors 00000079: timings 6.96s
Calib Neighbors 00000081: timings 7.16s
Calib Neighbors 00000082: timings 7.28s
Calib Neighbors 00000083: timings 7.39s
Calib Neighbors 00000085: timings 7.58s
Calib Neighbors 00000087: timings 7.75s
Calib Neighbors 00000089: timings 7.87s

neighborhood: [ 5 20 35 37 38]
Training set size: 2658
Test set size: 704
training start!!**

However, nothing happens after this stage and I think the problem is in this line of code in trainer.py
 **inputs = self.test_loader.__iter__().next()#test_loader_iter.next()
            print("Hello2")
            for k, v in inputs.items():  # load inputs to device.
                if type(v) == list:**
as the print("Hello2") statement does not show.
Do you know what the problem is and how I can solve this problem?
Thank you and I look forward to hearing from you soon

**Comment by edshkim98** [comment]
Is this because the code is calculating grid_subsampling for every point in files?
When I used the original author code in pytorch, I remember this process took really long time too.

**Comment by edshkim98** [comment]
Fixed it dont worry :)

**Comment by yangninghua** [comment]
Stuck for a long time
